File: db/java/app/src/main/java/com/example/sales/dto/ProductResponse.java
package com.example.sales.dto;

import com.example.sales.domain.model.Product;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 商品レスポンスDTO
 */
@Data
public class ProductResponse {

    private String productCode;
    private String fullName;
    private String name;
    private String kanaName;
    private Integer unitPrice;
    private Integer primeCost;
    private String supplierCode;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    /**
     * EntityからDTOへの変換
     */
    public static ProductResponse fromEntity(Product product) {
        ProductResponse response = new ProductResponse();
        response.setProductCode(product.getProductCode());
        response.setFullName(product.getProductFormalName());
        response.setName(product.getProductAbbreviation());
        response.setKanaName(product.getProductNameKana());
        response.setUnitPrice(product.getSellingPrice());
        response.setPrimeCost(product.getCostOfSales());
        response.setSupplierCode(product.getSupplierCode());
        response.setCreatedAt(product.getCreatedAt());
        response.setCreatedBy(product.getCreatedBy());
        response.setUpdatedAt(product.getUpdatedAt());
        response.setUpdatedBy(product.getUpdatedBy());
        return response;
    }
}

File: db/csharp/SalesManagement.Api/Dtos/ProductResponse.cs
using SalesManagement.Domain.Models;

namespace SalesManagement.Api.Dtos;

/// <summary>
/// 商品レスポンスDTO
/// </summary>
public class ProductResponse
{
    public string ProductCode { get; set; } = string.Empty;
    public string FullName { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string KanaName { get; set; } = string.Empty;
    public int UnitPrice { get; set; }
    public int PrimeCost { get; set; }
    public string SupplierCode { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public string CreatedBy { get; set; } = string.Empty;
    public DateTime UpdatedAt { get; set; }
    public string UpdatedBy { get; set; } = string.Empty;

    /// <summary>
    /// EntityからDTOへの変換
    /// </summary>
    public static ProductResponse FromEntity(Product product)
    {
        return new ProductResponse
        {
            ProductCode = product.ProductCode,
            FullName = product.ProductFormalName,
            Name = product.ProductAbbreviation,
            KanaName = product.ProductNameKana,
            UnitPrice = product.SellingPrice,
            PrimeCost = product.CostOfSales,
            SupplierCode = product.SupplierCode ?? string.Empty,
            CreatedAt = product.CreatedAt,
            CreatedBy = product.CreatedBy,
            UpdatedAt = product.UpdatedAt,
            UpdatedBy = product.UpdatedBy
        };
    }
}
